frontend/tests/LoginForm.test.tsx

```diff
@@ -0,0 +1,136 @@
+import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import LoginForm from "../src/components/LoginForm";
+import { login } from "../src/utils";
+
+// mock login to prevent backend fetch and get call data
+vi.mock("../src/utils", async (importOriginal) => {
+    return {
+        ...(await importOriginal<typeof import("../src/utils")>()),
+        login: vi.fn((enteredData: { email: string; password: string }) => [
+            true,
+            { id: 1, email: enteredData.email },
+        ]),
+    };
+});
```

Comment:
It might be a good idea to also test the scenario where the user enters incorrect login info.